### PR TITLE
Added a validation against minimum  group size validation.

### PR DESCRIPTION
**Pull request by nikomakela**
PT-798 Added the missing validation against minimum  group size validation. If the total count in under minimum required, same validation message will appear under both the study group count fields.

Issue: https://helsinkisolutionoffice.atlassian.net/browse/PT-798

Other validation cases here (in a linked task): https://helsinkisolutionoffice.atlassian.net/browse/PT-772?focusedCommentId=19250

### CODE REVIEW

**Comment by nikomakela** [comment]
Test shots from PR-server:  

image

image

image

File: src/domain/enrolment/enrolmentForm/ValidationSchema.tsx [hunk 1]
```diff
@@ -41,24 +41,43 @@ export default Yup.object().shape({
...
+          sizePair < 0 ||
+          sizePair < minGroupSize ||
+          sizePair > maxGroupSize
+        ) {
+          return (
+            schema
+              // Min-limit is current a gap to minimum group size.
+              .min(
+                sizePair != null ? minGroupSize - sizePair : minGroupSize,
+                (param) => ({
```
Comment (Sartonon) [inline src/domain/enrolment/enrolmentForm/ValidationSchema.tsx:62]: minor but params not used and can be deleted :)

File: src/domain/enrolment/enrolmentForm/ValidationSchema.tsx [hunk 2]
```diff
@@ -41,24 +41,43 @@ export default Yup.object().shape({
...
+          !sizePair ||
+          sizePair < 0 ||
+          sizePair < minGroupSize ||
+          sizePair > maxGroupSize
+        ) {
+          return (
+            schema
+              // Min-limit is current a gap to minimum group size.
+              .min(
+                sizePair != null ? minGroupSize - sizePair : minGroupSize,
```
Comment (Sartonon) [inline src/domain/enrolment/enrolmentForm/ValidationSchema.tsx:61]: typings suggest that sizePair cant be null or undefined so this logic is little bit strange

Comment (Sartonon) [reply src/domain/enrolment/enrolmentForm/ValidationSchema.tsx:61]: Typings could be improved. `sizePair: number | undefined`?

**Review by Sartonon — COMMENTED** [review]
Must admit that this logic is hard to grasp but code can't always be simple. Fortunately we have good tests and this seems to work :D

**Comment by nikomakela** [comment]
> Must admit that this logic is hard to grasp but code can't always be simple. Fortunately we have good tests and this seems to work :D

I agree! :) All this could be changed to a simpler solution where there would only be same validation messages for both the fields about not met min or max group size. However, I think that it would not be as informative for the user. But that's my suggestion for the next iteration of this validation saga :D

File: src/domain/enrolment/enrolmentForm/ValidationSchema.tsx (hunk 2 above)
Comment (nikomakela) [reply src/domain/enrolment/enrolmentForm/ValidationSchema.tsx:61]: Updated.

File: src/domain/enrolment/enrolmentForm/ValidationSchema.tsx (hunk 1 above)
Comment (nikomakela) [reply src/domain/enrolment/enrolmentForm/ValidationSchema.tsx:62]: Removed.

**Comment by Sartonon** [comment]
Hopefully this validation is enough for now :D

**Review by Sartonon — APPROVED** [review]
Very good! 👍

**Pull request merged**